analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/PublicMethodArgumentsShouldBeCheckedForNull.cs

```diff
@@ -406,3 +406,175 @@ public void ReassignedFromConstructorOut(out StringBuilder parameter)
 
     private StringBuilder Create() => null;
 }
+
+public class ClassAccessibility
+{
```

Comment:
Let's add "ShouldExecute"-specific UTs that will show that we don't raise for parameters of these:
```
                SyntaxKind.AnonymousMethodExpression,
                SyntaxKind.SimpleLambdaExpression,
                SyntaxKind.ParenthesizedLambdaExpression);
```
As the implementation requires `BaseMethodDeclarationSyntax`, it should just work.